src/tools/publish_toolstate.py

```diff
@@ -45,11 +56,48 @@ def read_current_status(current_commit, path):
                 return json.loads(status)
     return {}
 
+def issue(
+    tool,
+    maintainers,
+    relevant_pr_number,
+    relevant_pr_user,
+    pr_reviewer,
+):
+    # Open an issue about the toolstate failure.
+    gh_url = 'https://api.github.com/repos/rust-lang/rust/issues'
+    assignees = [x.strip() for x in maintainers.split('@') if x != '']
+    assignees.append(relevant_pr_user)
+    response = urllib2.urlopen(urllib2.Request(
```

Comment:
We might want to try `urlopen` error.

I'm not familiar with python. So I'm sorry if I have a something misunderstood.